psdisk/psdisk.c

```diff
@@ -359,227 +393,159 @@ static int psdisk_updatePartsList(ptable_partition_t *pHeaders, uint32_t pCnt)
 
 static int psdisk_updateImg(void)
 {
-	uint32_t pCnt;
-	ptable_partition_t *pHeaders;
+	ptable_t *ptable;
+	int err;
 
-	if ((pHeaders = ptable_readPartitions(&pCnt, &psdisk_common.mem)) == NULL) {
-		fprintf(stderr,"The file contains incorrect partition table.\n");
+	ptable = psdisk_readImg();
+	if (ptable == NULL) {
+		fprintf(stderr, "The file contains incorrect partition table.\n");
 		return -1;
 	}
 
-	if (psdisk_updatePartsList(pHeaders, pCnt) < 0) {
-		free(pHeaders);
+	err = psdisk_updatePartsList(ptable);
+	free(ptable);
+	if (err < 0) {
 		return -1;
 	}
 
 	if (psdisk_createImg() < 0) {
-		free(pHeaders);
 		return -1;
 	}
 
-	free(pHeaders);
-
 	return 0;
 }
 
 
 /* Parsing data from user */
 
-static int psdisk_parseToRm(void)
+static int psdisk_parseToRm(const char *arg)
 {
+	struct plist_node_t *node;
 	int i;
-	struct plist_node_t *pNode;
 
 	/* Parse partition name */
-	for (i = 0; i < strlen(optarg); ++i) {
-		if (!isalnum(optarg[i])) {
-			fprintf(stderr,"Invalid partition name - %s.\n", optarg);
-			return -1;
+	for (i = 0; i < sizeof(node->part.name); i++) {
+		if (arg[i] == '\0') {
+			break;
```

Comment:
Maybe `strlen` ?
```suggestion
	i = strlen(arg);
```
later you check `if ((i == 0) || (i >= sizeof(node->part.name)) {`.

Otherwise if the `for-loop` version is to stay, `i` is always less than `sizeof(node->part.name)` so the second check `|| (i >= sizeof(node->part.name)` is redundant.

Reply:
I'd like to stay with the `for-loop` version. It''ll check less characters than `strlen()` in case of a too long partition name.
The second `if` check is necessary for detecting too long partition name (it could be `i == sizeof(node->part.name)`).

Reply:
Ack.

Reply:
I don't really think that it hurts that much we might check more chars with strlen, but there's solution for this too:
`i = strnlen(arg, sizeof(node->part.name));`
pls don't reimplement standard functions.

Reply:
I forgot about `strnlen()`. Fixed in the next force-push.